Common/Projectiles/HitNPCHooks.cs

```diff
@@ -0,0 +1,47 @@
+﻿using Terraria.ModLoader.Core;
+
+namespace PathOfTerraria.Common.Projectiles;
+
+internal class HitNPCHooks : ILoadable
+{
+	/// <summary>
+	/// Allows projectiles to run code after the projectile(s) are hit.
+	/// </summary>
+	internal interface IPostHitNPCProjectile
+	{
+		private static readonly GlobalHookList<GlobalProjectile> Hook = ProjectileLoader.AddModHook(GlobalHookList<GlobalProjectile>.Create(i => ((IPostHitNPCProjectile)i).PostHitNPC));
```

Comment:
From an outside usage perspective, it makes more sense to nest the implementation inside the interface, rather than the interface inside the implementation. Nested ILoadables are valid.